app/src/gui/fields/TakePoint.tsx

```diff
@@ -56,89 +59,99 @@ interface Props {
   label?: string;
 }
 
-export class TakePoint extends React.Component<
-  FieldProps &
+export const TakePoint = (
+  props: FieldProps &
     Props &
     ButtonProps & {
       ValueTextProps: React.HTMLAttributes<HTMLSpanElement>;
       ErrorTextProps: React.HTMLAttributes<HTMLSpanElement>;
       NoErrorTextProps: React.HTMLAttributes<HTMLSpanElement>;
     }
-> {
-  getPositionOptions() {
+) => {
+  const [noPermission, setNoPermission] = React.useState(false);
+
+  const getPositionOptions = () => {
     return {
-      // override the default capacitor setting for enableHighAccuracy as we
-      // almost always want the high accuracy version, and users should only
-      // opt out when they understand when it's not needed
-      enableHighAccuracy: this.props.enableHighAccuracy ?? true,
-      // same default as capacitor
-      timeout: this.props.timeout ?? 10000, // in milliseconds
-      // same default as capacitor
-      maximumAge: this.props.maximumAge ?? 0, // in milliseconds
+      enableHighAccuracy: props.enableHighAccuracy ?? true,
+      timeout: props.timeout ?? 10000,
+      maximumAge: props.maximumAge ?? 0,
     };
-  }
+  };
 
-  async takePoint() {
+  const takePoint = async () => {
     try {
       const coordinates = capacitor_coordindates_to_faims_pos(
-        await Geolocation.getCurrentPosition(this.getPositionOptions())
+        await Geolocation.getCurrentPosition(getPositionOptions())
       );
-      this.props.form.setFieldValue(this.props.field.name, coordinates, true);
+      props.form.setFieldValue(props.field.name, coordinates, true);
     } catch (err: any) {
       logError(err);
-      this.props.form.setFieldError(this.props.field.name, err.message);
-    }
-  }
-  render() {
-    const pos = this.props.field.value;
-    const error = this.props.form.errors[this.props.field.name];
-    const instruction_text =
-      this.props.instruction_text ?? 'Click to save current location';
-    let positionText = <span>No point taken.</span>;
-    if (pos !== null && pos !== undefined && pos.geometry !== undefined) {
-      positionText = (
-        <span {...this.props['ValueTextProps']}>
-          Lat: {pos.geometry.coordinates[1] ?? 'Not captured'}; Long:{' '}
-          {pos.geometry.coordinates[0] ?? 'Not captured'}; Acc:{' '}
-          {pos.properties.accuracy ?? 'Not captured'}; Alt:{' '}
-          {pos.properties.altitude ?? 'Not captured'}; AltAcc:{' '}
-          {pos.properties.altitude_accuracy ?? 'Not captured'}
-        </span>
-      );
-    }
-    let error_text = <span {...this.props['NoErrorTextProps']}></span>;
-    if (error) {
-      error_text = (
-        <span {...this.props['ErrorTextProps']}>{error.toString()}</span>
-      );
+      setNoPermission(true);
```

Comment:
Is there any other source of error other than permissions? Is there a specific error code you could catch for this?